status/projects.py

```diff
@@ -167,6 +167,15 @@ def project_summary_data(self, row):
             except ValueError:
                 pass
 
+        #Try to fetch a name for the contact field, not just an e-mail
+        if 'contact' in row.value:
+            if 'owner' in row.value['order_details'] and row.value['contact'] == row.value['order_details']['owner']['email']:
+                row.value['contact'] = str(row.value['order_details']['owner']['name']) + ': ' + str(row.value['order_details']['owner']['email'])
+            elif 'fields' in row.value['order_details'] and row.value['contact']  == row.value['order_details']['fields']['project_lab_email']:
+                row.value['contact'] = str(row.value['order_details']['fields']['project_lab_name']) + ': ' + str(row.value['order_details']['fields']['project_lab_email'])
+            elif 'fields' in row.value['order_details'] and row.value['contact']  == row.value['order_details']['fields']['project_pi_email']:
```

Comment:
Combine the `'fields' in row.value['order_details']` conditions and make a nested if elif for the second condition so that its easier to understand later. You can also assign `row.value['order_details']['fields']` to a variable